apps/crud/src/app/components/item.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, Input } from '@angular/core';
+import { Todo } from '../models/interface.todo';
+import { TodosService } from '../services/todos.service';
+import { TodosStore } from '../store/todos.store';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+import { CommonModule } from '@angular/common';
+
+@Component({
+  standalone: true,
+  selector: 'app-item',
+  imports: [CommonModule, MatProgressSpinnerModule],
+  template: `<div>
+      <p>{{ todo.title }}</p>
+      <button (click)="update(todo)" [disabled]="todo.id === processingItemId">
+        Update
+      </button>
+      <button (click)="delete(todo)" [disabled]="todo.id === processingItemId">
+        Delete
+      </button>
+    </div>
+    <div *ngIf="loadingText">
+      <p>{{ loadingText }}</p>
+      <mat-spinner></mat-spinner>
+    </div> `,
+  styles: [],
+  providers: [],
+})
+export class ItemComponent {
+  @Input() todo!: Todo;
+  @Input() processingItemId: number | null = null;
+  loadingText = '';
+
+  constructor(
+    private todosService: TodosService,
+    private todosStore: TodosStore
+  ) {}
+
+  update(todo: Todo) {
+    this.loadingText = 'Updating...';
+    this.todosStore.startProcessing(todo.id);
+    this.todosService.updateTodo(todo).subscribe((updatedTodo) => {
+      this.loadingText = '';
+      this.todosStore.stopProcessing();
+      this.todosStore.updateTodoInList(updatedTodo);
+    });
+  }
+
+  delete(todo: Todo) {
+    this.loadingText = 'Deleting...';
+    this.todosStore.startProcessing(todo.id);
+    this.todosService.deleteTodo(todo).subscribe(() => {
+      this.loadingText = '';
+      this.todosStore.stopProcessing();
+      this.todosStore.removeTodoFromList(todo.id);
+    });
+  }
+}
```

Comment:
all this code is imperative. Try to avoid this at best. This challenge has been created to practice declarative and reactive coding. 

You should always avoid setting properties inside a manual subscribe

Better way is to update your store and to listen to your local state and react to it. Not update the state and some properties inside your component. That's very dangerous because your state can be different than your local properties